rubin_jupyter_utils/hub/scanrepo/scanrepo.py

```diff
@@ -143,75 +141,188 @@ def _read_cachefile(self):
+            #  cachefile
+            self.logger.debug("Updating last_scan from cachefile mtime.")
+            self.last_scan = datetime.datetime.fromtimestamp(
+                os.path.getmtime(fn)
+            )
 
     def _describe_tag(self, tag):
-        # New-style tags have underscores separating components.
-        # Don't log it; way too noisy.
+        # This method started as a way to extract a human-friendly description
+        #  from the tag name; it has been extended to extract a tag-derived
+        #  semantic version from the name.  This will generally look like
+        #  a date for non-release versions.
+        #
+        # Specifically, dailies and weeklies, and experimentals built
+        #  from either of those, will have a known date, because it's
+        #  in the tag (assuming that weeklies are always built on
+        #  Saturday, which is generally a good assumption), and
+        #  therefore we can easily create a semantic version for each of
+        #  those (major/minor/patch) that corresponds to year/month/day, and
+        #  that will sort them in production order.
+        #
+        # In general we don't build dailies on Saturday, so version collison
+        #  will be rare, and furthermore the only place we'd have dailies,
+        #  weeklies, and experimentals made from those all jammed together
+        #  will be the all-tags dropdown.  In which case, well, maybe a
+        #  weekly and daily will be reversed relative to their actual
+        #  production order, but who really cares?
+        #
+        # Releases have numbers that let us assign a sort order to them.
+        #  The question then becomes what to do with them relative to all
+        #  other builds.  Fortunately this only matters in the all-tags
+        #  dropdown; so I propose to just sort them at the end of the list.
+        #  The easiest way to do this is to make *their* version just the
+        #  semver of the release tags.  That means that releases will all
+        #  have a much lower version than non-release builds.  If we don't
+        #  like that, we can make their major version 9999, and tack the
+        #  rest on after that.
+        #
+        # For experimental and resolved ("recommended" or "latest") tags
+        #  we call this recursively, and fix up the descriptions and
+        #  version components when we return from the inner call.
+        #
+        # New-style tags (since early 2019) have underscores separating
+        #  components.
+        #
+        # This method is called far too often to log with eliot, because
+        #  we pop in and out of it many, many times when scanning a repo.
+        #
         ld = tag  # Default description is just the tag name
         components = None
-        if tag.find("_") != -1:
+        ttype = None
+        major = 0
+        minor = 0
+        patch = 0
+        rest = None
+        if tag.find("_") != -1:  # New-style tag
```

Comment:
There must be someway less c-like to write this for clarity - 
`if ‘_’ in tag:`
or something?

Reply:
That is indeed the less c-like way to write it.  Changed.